File: Data Structures & Algorithms/is-palindrome/submission-11.java
class Solution {
    public boolean isPalindrome(String s) {
        int l=0;
        int r=s.length()-1;
        while (r>l){
            if (!Character.isLetterOrDigit(Character.toLowerCase(s.charAt(r)))){
                r--;
            }
            else if (!Character.isLetterOrDigit(Character.toLowerCase(s.charAt(l)))){
                l++;
            }
            else{
                if (Character.toLowerCase(s.charAt(r))!=Character.toLowerCase(s.charAt(l))){
                    return false;
                }
                l++;
                r--;
            }

        }
        return true;
    }
}

File: Data Structures & Algorithms/is-anagram/submission-11.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        c={}
        d={}
        for i in s:
            c[i]=c.get(i,0)+1
        for i in t:
            d[i]=d.get(i,0)+1
        return c==d